### PR TITLE
Name lookup error on retry

### PR DESCRIPTION
**Pull request by euri10**
> We indeed want to retry name resolution errors

humble attempt at solving the above, I'm not sure if I correctly understood the original intent, in my mind it translates into the diff below:
```diff
+        elif isinstance(err, NameResolutionError):
+            return True
```

I also hesitated a lot where to add some tests, and while checking where that `_is_connection_error` function was called it seemed logical to put in the proxymanager tests where retry are used a lot, but I'm certain there might be other places where this could be interesting to have as well, I'm not familiar enough with the code base so I'd appreciate pointers on that if that's required

### CODE REVIEW

File: src/urllib3/util/retry.py [hunk 1]
```diff
@@ -344,6 +345,8 @@ def _is_connection_error(self, err: Exception) -> bool:
         """
         if isinstance(err, ProxyError):
             err = err.original_error
+        elif isinstance(err, NameResolutionError):
+            return True
         return isinstance(err, ConnectTimeoutError)
```
Comment (sethmlarson) [inline src/urllib3/util/retry.py:350]: Thoughts on `return isinstance(err, (ConnectTimeoutError, NameResolutionError))`?

File: src/urllib3/exceptions.py [hunk 2]
```diff
@@ -382,6 +382,9 @@ class UnrewindableBodyError(HTTPError):
 class NameResolutionError(HTTPError, socket.gaierror):
     """Raised when host name resolution fails."""
 
-    def __init__(self, host: str, reason: socket.gaierror):
-        message = f"Failed to resolve '{host}' ({reason})"
-        HTTPError.__init__(self, message)
+    original_error: Exception
+
+    def __init__(self, host: str, error: Exception) -> None:
```
Comment (sethmlarson) [inline src/urllib3/exceptions.py:387]: Not sure if we need `original_error` on this exception?

File: src/urllib3/util/retry.py (hunk 1 above)
Comment (euri10) [reply src/urllib3/util/retry.py]: funny how when I see this I felt like it was so obvious I dont get why I did not write it in the first place

File: src/urllib3/exceptions.py (hunk 2 above)
Comment (euri10) [reply src/urllib3/exceptions.py]: I removed it

**Comment by pquentin** [comment]
Closing per discussion in #1003

**Pull request closed without merging**